### PR TITLE
[refact](meta-service) Split `commit_txn` into several functions

### PR DESCRIPTION
**Pull request by swjtu-zhanglei**
`commit_txn` code is too long, so split it into several subfunctions

### CODE REVIEW

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

File: cloud/src/meta-service/meta_service.h [hunk 1]
```diff
@@ -280,6 +276,18 @@ class MetaServiceImpl : public cloud::MetaService {
             const AlterInstanceRequest* request,
             std::function<std::pair<MetaServiceCode, std::string>(InstanceInfoPB*)> action);
 
+    void commit_txn_immediately(
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service.h:279]: hide it to .cpp file

File: cloud/src/meta-service/meta_service_txn.cpp [hunk 2]
```diff
@@ -772,17 +732,129 @@ void MetaServiceImpl::commit_txn(::google::protobuf::RpcController* controller,
...
+    int64_t db_id;
+    std::vector<std::pair<std::string, doris::RowsetMetaCloudPB>> tmp_rowsets_meta;
+    scan_tmp_rowset(instance_id, txn_id, txn_kv_, code, msg, ss, &db_id, &tmp_rowsets_meta);
+    if (code != MetaServiceCode::OK) {
+        LOG(WARNING) << "scan_tmp_rowset failed, txn_id=" << txn_id << " code=" << code;
+        return;
+    }
+    commit_txn_immediately(controller, request, response, done, code, msg, ss, instance_id, db_id,
+                           tmp_rowsets_meta);
+    return;
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_txn.cpp:781]: redundant return;

**Comment by SWJTU-ZhangLei** [comment]
run buildall

File: cloud/src/meta-service/meta_service.h [hunk 3]
```diff
@@ -279,7 +275,6 @@ class MetaServiceImpl : public cloud::MetaService {
     std::pair<MetaServiceCode, std::string> alter_instance(
             const AlterInstanceRequest* request,
             std::function<std::pair<MetaServiceCode, std::string>(InstanceInfoPB*)> action);
```
Comment (w41ter) [inline cloud/src/meta-service/meta_service.h:277]:
```suggestion
            std::function<std::pair<MetaServiceCode, std::string>(InstanceInfoPB*)> action);

```

**Review by w41ter — DISMISSED** [review]

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

**Review by gavinchou — APPROVED** [review]

**Pull request merged**